src/test/java/io/axoniq/axonserver/connector/event/impl/PersistentEventChannelImplTest.java

```diff
@@ -0,0 +1,94 @@
+package io.axoniq.axonserver.connector.event.impl;
+
+import io.axoniq.axonserver.connector.AxonServerConnectionFactory;
+import io.axoniq.axonserver.connector.ResultStreamPublisher;
+import io.axoniq.axonserver.connector.event.EventChannel;
+import io.axoniq.axonserver.connector.event.PersistedStreamProperties;
+import io.axoniq.axonserver.connector.event.PersistentStream;
+import io.axoniq.axonserver.grpc.event.Event;
+import io.axoniq.axonserver.grpc.event.EventWithToken;
+import org.junit.jupiter.api.*;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Flux;
+import reactor.core.scheduler.Schedulers;
+
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+class PersistentEventChannelImplTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(PersistentEventChannelImplTest.class);
+    private EventChannel eventChannel;
+
+    @BeforeEach
+    void setup() throws ExecutionException, InterruptedException {
+        AxonServerConnectionFactory connectionFactory = AxonServerConnectionFactory.forClient("demo",
```

Comment:
This test seems to assume AxonServer is running on port 8124 to run. Could we use testcontainers instead?